TIDALDL-PY/tidal_dl/download.py

```diff
@@ -93,13 +90,25 @@ def __setMetaData__(track, album, filepath, contributors, lyrics):
     obj.title = track.title
     if not aigpy.string.isNull(track.version):
         obj.title += ' (' + track.version + ')'
-    obj.artist = __getArtists__(track.artists)
+    # required until https://github.com/AIGMix/AIGPY/pull/3/files
+    if '.mp4' in filepath or '.m4a' in filepath:
+        obj.artist = []
+        for artist in track.artists:
+            obj.artist.append(artist.name)
+    else:
+        obj.artist = __getArtists__(track.artists)

```

Comment:
do not modify the aigpy interface

Reply:
it's not necessary to modify the aigpy interface, but aigpy only works with list for mp4, and only works with string for flac and mp3

if saving flac or mp3 as list, it only saves the first artist

this fix stores artists as a string for mp3 and flac because that is what aigpy needs

Reply:
if you do want to fix the aigpy interface, you should use a major version bump because it would be a breaking change and you don't want people to accidentally upgrade to a different interface